### ISSUE TITLE
Attempt to invoke interface method 'int java.lang.CharSequence.length()' on a null object reference

### ISSUE THREAD

**Issue by fontesrp** [facebook/react-native#28967]
## Description

We are getting the following error report on Sentry for our Android production build:

**NullPointerException** android.text.SpannableStringBuilder in append
Attempt to invoke interface method 'int java.lang.CharSequence.length()' on a null object reference

```
java.lang.NullPointerException: Attempt to invoke interface method 'int java.lang.CharSequence.length()' on a null object reference
    at android.text.SpannableStringBuilder.append(SpannableStringBuilder.java:269)
    at com.facebook.react.views.text.ReactBaseTextShadowNode.buildSpannedFromShadowNode(ReactBaseTextShadowNode.java:113)
    at com.facebook.react.views.text.ReactBaseTextShadowNode.spannedFromShadowNode(ReactBaseTextShadowNode.java:260)
    at com.facebook.react.views.text.ReactTextShadowNode.onBeforeLayout(ReactTextShadowNode.java:243)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:873)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.notifyOnBeforeLayoutRecursive(UIImplementation.java:871)
    at com.facebook.react.uimanager.UIImplementation.updateViewHierarchy(UIImplementation.java:634)
    at com.facebook.react.uimanager.UIImplementation.dispatchViewUpdates(UIImplementation.java:600)
    at com.facebook.react.uimanager.UIManagerModule.onBatchComplete(UIManagerModule.java:803)
    at com.facebook.react.bridge.NativeModuleRegistry.onBatchComplete(NativeModuleRegistry.java:119)
    at com.facebook.react.bridge.CatalystInstanceImpl$BridgeCallback.onBatchComplete(CatalystInstanceImpl.java:172)
    at com.facebook.react.bridge.queue.NativeRunnable.run(NativeRunnable.java)
    at android.os.Handler.handleCallback(Handler.java:751)
    at android.os.Handler.dispatchMessage(Handler.java:95)
    at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
    at android.os.Looper.loop(Looper.java:154)
    at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:226)
    at java.lang.Thread.run(Thread.java:761)
```

The same issue was reported on https://github.com/facebook/react-native/issues/22815, but it was closed because their RN version was too old.

## React Native version:

```
System:
    OS: macOS 10.15.4
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 25.34 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.16.3 - /var/folders/ts/48vtw6y57470j_gnv8wx20gh0000gp/T/yarn--1590434740439-0.22439665870933068/node
    Yarn: 1.22.4 - /var/folders/ts/48vtw6y57470j_gnv8wx20gh0000gp/T/yarn--1590434740439-0.22439665870933068/yarn
    npm: 6.14.5 - ~/.nvm/versions/node/v12.16.3/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.9.2 - /Users/rfontes/.rvm/gems/ruby-2.6.3/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 13.5, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
    Android SDK:
      API Levels: 28, 29
      Build Tools: 28.0.3, 29.0.3
      System Images: android-28 | Intel x86 Atom_64, android-28 | Google APIs Intel x86 Atom_64, android-28 | Google Play Intel x86 Atom_64, android-29 | Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom_64, android-29 | Google Play Intel x86 Atom_64
      Android NDK: Not Found
  IDEs:
    Android Studio: 3.6 AI-192.7142.36.36.6392135
    Xcode: 11.5/11E608c - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_252 - /Users/rfontes/.sdkman/candidates/java/current/bin/javac
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.11.0 => 16.11.0 
    react-native: 0.62.2 => 0.62.2 
  npmGlobalPackages:
    *react-native*: Not Found
```

## Steps To Reproduce

We are not able to reproduce the error. It only appears as an error report in production.

## Expected Results

A safe-check should be added for when the text value of a child `ReactRawTextShadowNode` is `null`.

## Snack, code example, screenshot, or link to a repository:

The error is being thrown by this [line](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/text/SpannableStringBuilder.java#270) of the `SpannableStringBuilder` class because RN is passing a `null` value to its `append` method in [this](https://github.com/facebook/react-native/blob/ce0edca620d49597aa73c71a68b70efa170bc828/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactBaseTextShadowNode.java#L115) call.

I suggest adding a safe-check before calling `append`. Something like this:

```java
      if (child instanceof ReactRawTextShadowNode) {
        String childText = TextTransform.apply(((ReactRawTextShadowNode) child).getText(), textAttributes.getTextTransform());
        if (childText != null) {
          sb.append(childText);
        }
      }
```

**Comment by fabriziobertoglio1987** [comment]
I need to see your application code, specific responsible for triggering the error.

I believe this is connected to `Text` component 

https://github.com/heimashi/React-Native-Android-Demo/blob/master/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextView.java

**Comment by fabriziobertoglio1987** [comment]
why

```java
TextTransform.apply(
    ((ReactRawTextShadowNode) child).getText(), textAttributes.getTextTransform());
```
returns `null` ?

**Comment by fontesrp** [comment]
Hello @fabriziobertoglio1987, thanks for the response!

So, as I said, we are not able to reproduce the error. I tried a few scenarios in your snack, but had no success. I opened this report because the method `getText` of `ReactRawTextShadowNode` [is nullable](https://github.com/facebook/react-native/blob/9d6d39de86057f92d41a9f7eed7fdddf1ac09582/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactRawTextShadowNode.java#L34) and [so is](https://github.com/facebook/react-native/blob/9d6d39de86057f92d41a9f7eed7fdddf1ac09582/ReactAndroid/src/main/java/com/facebook/react/views/text/TextTransform.java#L22) the method `apply` of `TextTransform`; and they are being passed to `append` of `SpannableStringBuilder` which assumes its argument is non-null (even if it's not annotated). This could be considered a bug, or at least a malpractice, in RN's source code. However, I understand if the lack of a reproducible example would prevent you from submitting a PR. Maybe someone else here can help with this point.

I can't see how checking for null on JS would help, as your snack shows that a Text tag with null children will not cause the bug. I'm not sure we're allowed to share our source code, but the (very) low number of users experiencing the problem leads us to believe that it's caused by a combination of some very specific cellphone state with faulty API responses.

One last thing I'd like to add, and correct me if I'm wrong here, is that since the text content of the tag is null, not calling `sb.append` would have the same effect as not rendering the tag, which is the user not seeing any text. Therefore I don't think we would be hiding the bug by adding the check on Java, but simply preventing the app from closing when this state arises.

Thanks again!
